### ISSUE TITLE
Create and insert block if not present according to block template

### ISSUE THREAD

**Issue by estebanwaseaten** [WordPress/gutenberg#15593]
I am moving from old-style meta boxes to a "locked block template" scenario. For new posts this works very well. But already existing posts do not contain the block according to the template. I am having trouble automatically inserting this block (if not present):
So far I tried to register a plugin via wp.plugins.registerPlugin that checks if the block is present and adds it if not via:
`wp.data.dispatch('core/editor').insertBlock( wp.blocks.createBlock( 'namespace/name' )` and
`wp.data.dispatch( 'core/editor').savePost()`
within a anonymous function subscribed to wp.data.: `wp.data.subscribe(...`
this at least renders the block correctly in the visual editor of gutenberg, but it never really adds any code (frontend or the code editor of gutenberg). 

also there is a javascript error generated: "Cannot commit transaction with id "post-update" because it does not exist"

Thank you for any hints/directions on where and how to check for blocks and add them...

[pc_admin_sidebar_publications.txt](https://github.com/WordPress/gutenberg/files/3172239/pc_admin_sidebar_publications.txt)

**Comment by youknowriad** [comment]
`wp.data.dispatch('core/editor').insertBlock( wp.blocks.createBlock( 'namespace/name' ) )` This works AFAIK. You just need to call it at the right time though (once everything is loaded properly)

**Comment by estebanwaseaten** [comment]
I stumbled over this problem again. I get the "The content of your post doesn’t match the template assigned to your post type." warning, whenever I open an old post that does not have the right blocks according to the block template and I can click "Reset the template". Manually this works perfectly well.

I would like to do this automatically as soon as the post loads. Therefore I registered and enqueued the following simple .js:

`( function( wp ) {
        var once = 0;
        wp.data.subscribe( test );
               
        function test()
        {
            var valid = wp.data.select( 'core/editor' ).isValidTemplate();
            if( !valid && (once == 0) )
            {
                console.log( 'invalid template - try to fix it... ' );
                once += 1;
                wp.data.dispatch( 'core/editor' ).synchronizeTemplate();
                wp.data.dispatch( 'core/editor' ).savePost();
            }
        }
} )( window.wp );`

this SEEMS to work. (The missing block appears due to synchronizeTemplate() and the Post updated. View Post message appears as if I had saved the post). 
But if I reload the post I get the same behavior (i.e. the post was not saved). If I fix the template manually everything works fine... what am I missing here?
Thanks very much in advance!
